kolibri/core/content/views.py

```diff
@@ -27,6 +30,35 @@ def _add_access_control_headers(request, response):
         response["Access-Control-Allow-Headers"] = requested_headers
 
 
+def get_referrer_url(request):
+    if request.META.get('HTTP_REFERER'):
+        # If available user HTTP_REFERER to infer the host as that will give us more
+        # information if Kolibri is behind a proxy.
+        return urlparse(request.META.get('HTTP_REFERER'))
+
+
+def generate_image_prefix_url(zipped_filename, parsed_referrer_url):
+    # Remove trailing slash
+    zipcontent = reverse(
+        'kolibri:core:zipcontent',
+        kwargs={
+            "zipped_filename": zipped_filename,
+            "embedded_filepath": ''
+        })[:-1]
+    if parsed_referrer_url:
+        # Reconstruct the parsed URL using only the scheme(0) and host + port(1)
+        zipcontent = urlunparse((parsed_referrer_url[0], parsed_referrer_url[1], zipcontent, '', '', ''))
+    return zipcontent.encode()
+
+
+def get_host(request, parsed_referrer_url):
+    if parsed_referrer_url:
+        host = urlunparse((parsed_referrer_url[0], parsed_referrer_url[1], '', '', '', ''))
+    else:
+        host = request.build_absolute_uri(OPTIONS['Deployment']['URL_PATH_PREFIX'])
```

Comment:
If the page is refreshed or reloaded, will the parsed_referrer_url still be set? If not, do we want to have some sort of guard for the `http://kolibri` case and revert to the old behavior of not adding a host to the URL when it happens? That way, at least for some other browsers that handle the `http:/zipcontent/` URLs will continue to work in this case.

Reply:
Yes, all requests to fetch JSON are done from within an XHR request, rather than a direct page load, so the header should always be set.

What you are suggesting is what I have done - if the referrer header is not set (which it will not be on e.g. a direct page load) then it falls back to the old behaviour.

Reply:
Calling `build_absolute_uri` was being called before when setting the `host` variable for the `Content-Security-Policy` header, so that part is the same, but it was not being used when doing the `content_with_path` replacement, so I wanted to double-check that `build_absolute_uri` wouldn't ever return a bad host name in this case. (As if it did, `http:/zipcontent/...` will at least work in some browsers, whereas `http://kolibri/zipcontent/...` won't work anywhere.) Very possible you learned a way to ensure `get_absolute_uri` and `get_host` don't return `http://kolibri/` after our discussion on Slack, but just wanted to double-check it.

Reply:
This function is only generating the host for the CSP header. See the `generate_image_url_prefix` function for the behaviour for image placeholder substitution URL generation - there the old behaviour is preserved, we just then add the previously defined path to the schema + host of the referer header if available.

Reply:
Gah, sorry, I'm blind. :( Thanks for clarifying!